docs/asdf/extending/compressors.rst

```diff
@@ -0,0 +1,57 @@
+.. currentmodule:: asdf.extension
+
+.. _extending_compressors:
+
+========================
+Binary block compressors
```

Comment:
@lgarrison this is the section I added on compressors, it's a little cursory but can you let me know if there are any errors or major omissions?  I also updated the Compressor class's docstrings after comparing them to the built-in compressor implementations and removed the "decompressors" property from ManifestExtension which I think is no longer needed.